File: flutter-app/cake_shop/lib/models/cake.dart
class Cake {
  final String id;
  final String name;
  final String description;
  final String category;
  final double basePrice;
  final String imageUrl;
  final List<String> flavors;
  final List<String> sizes;
  final double rating;
  final bool inStock;

  Cake({
    required this.id,
    required this.name,
    required this.description,
    required this.category,
    required this.basePrice,
    required this.imageUrl,
    required this.flavors,
    required this.sizes,
    required this.rating,
    required this.inStock,
  });

  factory Cake.fromJson(Map<String, dynamic> json) {
    return Cake(
      id: json['id'] as String,
      name: json['name'] as String,
      description: json['description'] as String? ?? '',
      category: json['category'] as String,
      basePrice: (json['basePrice'] as num).toDouble(),
      imageUrl: json['imageUrl'] as String? ?? '',
      flavors: List<String>.from(json['flavors'] ?? []),
      sizes: List<String>.from(json['sizes'] ?? []),
      rating: (json['rating'] as num?)?.toDouble() ?? 0,
      inStock: json['inStock'] as bool? ?? true,
    );
  }
}

File: backend/src/models/Cake.js
const mongoose = require('mongoose');

const cakeSchema = new mongoose.Schema(
  {
    name: { type: String, required: true, trim: true },
    description: { type: String, default: '' },
    category: {
      type: String,
      enum: ['BIRTHDAY', 'WEDDING', 'CUPCAKE', 'CUSTOM', 'SEASONAL'],
      required: true,
    },
    basePrice: { type: Number, required: true },
    imageUrl: { type: String, default: '' },
    flavors: { type: [String], default: [] },
    sizes: { type: [String], default: [] },
    rating: { type: Number, default: 0 },
    inStock: { type: Boolean, default: true },
  },
  { timestamps: true }
);

cakeSchema.methods.toPublicJSON = function () {
  return {
    id: this._id.toString(),
    name: this.name,
    description: this.description,
    category: this.category,
    basePrice: this.basePrice,
    imageUrl: this.imageUrl,
    flavors: this.flavors,
    sizes: this.sizes,
    rating: this.rating,
    inStock: this.inStock,
    createdAt: this.createdAt?.getTime(),
  };
};

module.exports = mongoose.model('Cake', cakeSchema);
